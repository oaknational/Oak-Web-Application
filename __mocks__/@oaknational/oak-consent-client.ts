const consentClientMock = jest.requireActual("@oaknational/oak-consent-client");

/**
 * Provides a mock implementation of OakConsentClient so that tests
 * do not try to hit the network
 */
class MockOakConsentClient {
  getState() {
    return {
      policyConsents: [],
      requireInteraction: false,
    };
  }
  logConsents() {
    return Promise.resolve();
  }
  getConsent() {
    return "pending";
  }
  onStateChange() {
    return () => {};
  }
}

module.exports = {
  ...consentClientMock,
  OakConsentClient: MockOakConsentClient,
};
