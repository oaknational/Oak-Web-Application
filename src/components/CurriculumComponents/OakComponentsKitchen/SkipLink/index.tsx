import {
  OakSecondaryButton,
  OakSecondaryButtonProps,
} from "@oaknational/oak-components";
import { ReactNode, useState } from "react";

const SkipLink = (
  props: Omit<OakSecondaryButtonProps, "element"> & {
    children: ReactNode;
    href: string;
  },
) => {
  const [isFocussed, setIsFocussed] = useState(false);
  const { children, ...buttonProps } = props;

  return (
    <OakSecondaryButton
      {...buttonProps}
      data-testid="skip-link"
      element="a"
      onFocus={() => setIsFocussed(true)}
      onBlur={() => setIsFocussed(false)}
      style={isFocussed ? {} : { position: "absolute", left: "-10000px" }}
    >
      {children}
    </OakSecondaryButton>
  );
};

export default SkipLink;
