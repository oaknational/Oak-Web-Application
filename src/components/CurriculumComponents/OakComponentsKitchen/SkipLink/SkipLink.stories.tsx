import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";
import { Meta, StoryObj } from "@storybook/nextjs";

import Component from ".";

const meta: Meta<typeof Component> = {
  component: Component,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Component>;

const TestComponent = () => {
  return (
    <OakThemeProvider theme={oakDefaultTheme}>
      <p>There is a skip link under this paragraph</p>
      <Component href="#testing">Skip to content</Component>
      <p>There is a skip link above this paragraph</p>
    </OakThemeProvider>
  );
};

export const SkipLink: Story = {
  render: TestComponent,
};
