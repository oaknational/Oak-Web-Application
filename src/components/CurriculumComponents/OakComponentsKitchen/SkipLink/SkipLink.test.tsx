import { act } from "react-dom/test-utils";
import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";

import SkipLink from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("SkipLink", () => {
  test("functions as expected", async () => {
    const { baseElement, getByTestId } = renderWithTheme(
      <OakThemeProvider theme={oakDefaultTheme}>
        <SkipLink href="foo">test</SkipLink>
      </OakThemeProvider>,
    );

    const element = getByTestId("skip-link");
    expect(element.style.position).toEqual("absolute");
    expect(element.style.left).toEqual("-10000px");

    act(() => {
      element.focus();
    });

    expect(element.style.position).toEqual("");
    expect(element.style.left).toEqual("");
    expect(baseElement).toMatchSnapshot();
  });
});
