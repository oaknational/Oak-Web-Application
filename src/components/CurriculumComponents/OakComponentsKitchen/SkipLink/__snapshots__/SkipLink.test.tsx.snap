// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink functions as expected 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-gmPhUn ifdHCZ fGXRhP"
    >
      <div
        class="sc-fqkvVR gIPXKL grey-shadow"
      />
      <div
        class="sc-fqkvVR gIPXKL yellow-shadow"
      />
      <a
        class="sc-jxOSlx sc-gFAWRd jEGIrQ jSchQQ internal-button"
        data-testid="skip-link"
        href="foo"
        style=""
      >
        <div
          class="sc-fqkvVR sc-gsFSXq bjeqlm dzOlKm"
        >
          <span
            class="sc-kpDqfm izJBLj"
          >
            test
          </span>
        </div>
      </a>
    </div>
  </div>
</body>
`;
